apps/src/lib/node/ledger/ethereum_node/events.rs

```diff
@@ -67,6 +42,106 @@ pub mod eth_events {
 
     pub type Result<T> = std::result::Result<T, Error>;
 
+    /// This represents all possible event types that can be emitted by any of
+    /// the Ethereum bridge smart contracts
+    #[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+    pub enum EventType {
+        Bridge(BridgeEventType),
+        Governance(GovernanceEventType),
+    }
+
+    #[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+    pub enum BridgeEventType {
+        TransferToNamada,
+        TransferToEthereum,
+    }
+
+    #[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+    pub enum GovernanceEventType {
+        ValidatorSetUpdate,
+        NewContract,
+        UpgradedContract,
+        UpdateBridgeWhiteList,
+    }
+
+    impl EventType {
+        pub fn all() -> Vec<Self> {
+            // TODO: we can get rid of this with enum_iterator::Sequence once we
+            // upgrade to Rust >= 1.63
+            vec![
+                EventType::Bridge(BridgeEventType::TransferToNamada),
+                EventType::Bridge(BridgeEventType::TransferToEthereum),
+                EventType::Governance(GovernanceEventType::ValidatorSetUpdate),
+                EventType::Governance(GovernanceEventType::NewContract),
+                EventType::Governance(GovernanceEventType::UpgradedContract),
+                EventType::Governance(
+                    GovernanceEventType::UpdateBridgeWhiteList,
+                ),
+            ]
```

Comment:
I was trying to avoid having something like this - there is a nice macro for deriving an iterator over all variants of an in commit https://github.com/anoma/namada/pull/566/commits/7417daa8ad15fe9b2294b4dd1fad956a5346c72e , but it requires Rust >=1.63 , so we'd need to wait to update to use it